### PR TITLE
Enable ``strict_equality`` mypy option

### PR DESCRIPTION
**Pull request by nsoranzo**
Also fix the 2 mypy errors reported when enabling this option.

## How to test the changes?
(Select all options that apply)
- [ ] I've included appropriate [automated tests](https://docs.galaxyproject.org/en/latest/dev/writing_tests.html).
- [x] This is a refactoring of components with existing test coverage.
- [ ] Instructions for manual testing are as follows:
  1. [add testing steps and prerequisites here if you didn't write automated tests covering all your changes]

## License
- [x] I agree to license these and all my past contributions to the core galaxy codebase under the [MIT license](https://opensource.org/licenses/MIT).

### CODE REVIEW

**Comment by jmchilton** [comment]
Informative! And a couple of solid bug fixes - thanks so much @nsoranzo!

**Review by jmchilton — APPROVED** [review]

File: lib/galaxy/model/__init__.py [hunk 1]
```diff
@@ -3915,16 +3917,26 @@ def extra_files_path_name(self):
...
         if self.external_filename:
             try:
                 return os.path.getsize(self.external_filename)
             except OSError:
                 return 0
-        else:
+        elif self.object_store:
             return self.object_store.size(self)
+        else:
+            return 0
```
Comment (mvdbeek) [inline lib/galaxy/model/__init__.py]: Shouldn't that be an error condition ?

File: lib/galaxy/objectstore/s3.py [hunk 2]
```diff
@@ -385,9 +385,8 @@ def _get_transfer_progress(self):
     def _get_size_in_s3(self, rel_path):
         try:
             key = self._bucket.get_key(rel_path)
-            if key:
-                return key.size
-        except S3ResponseError:
+            return key.size
+        except Exception:
```
Comment (mvdbeek) [inline lib/galaxy/objectstore/s3.py]: If it's not one of the expected errors it should probably not be caught ?

Comment (nsoranzo) [reply lib/galaxy/objectstore/s3.py]: This was to have the same logging and return value when `key` is `None`. I can change it to:

```suggestion
        except (S3ResponseError, AttributeError):
```

if that works for you?

File: lib/galaxy/model/__init__.py (hunk 1 above)
Comment (nsoranzo) [reply lib/galaxy/model/__init__.py]: Addressed in a rebase by using `assert self.object_store` instead.

File: lib/galaxy/objectstore/s3.py (hunk 2 above)
Comment (nsoranzo) [reply lib/galaxy/objectstore/s3.py]: Rebased with the suggested change.

**Review by mvdbeek — APPROVED** [review]
Thank you!

**Pull request merged**